OpenDreamClient/Resources/ResourceTypes/DreamResource.cs

```diff
@@ -18,4 +19,8 @@ public class DreamResource {
     public void WriteTo(Stream stream) {
         stream.Write(Data);
     }
+
+    public virtual void UpdateData(byte[] data) {
```

Comment:
Why are the update callbacks being called in DreamResourceManager and not here?

Reply:
because DMIResource overrides this and does more behaviour after the call that is needed before the update  callback is triggered.